src/Compression/CompressionCodecQATZSTD.cpp

```diff
@@ -0,0 +1,114 @@
+#ifdef ENABLE_QATZSTD_COMPRESSION
+#include <Compression/CompressionInfo.h>
+#include <Compression/CompressionFactory.h>
+#include <zstd.h>
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+#include "qatseqprod.h"
+#include "CompressionCodecZSTD.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int ILLEGAL_SYNTAX_FOR_CODEC_TYPE;
+    extern const int ILLEGAL_CODEC_PARAMETER;
+}
+
+class CompressionCodecQATZSTD : public CompressionCodecZSTD
```

Comment:
I guess, we should have a setting similar to existing setting `enable_deflate_qpl_codec` for this codec

Reply:
Done